app/controllers/admin/cafe_works_controller.rb

```diff
@@ -0,0 +1,116 @@
+# encoding:UTF-8
+class Admin::CafeWorksController < ApplicationController
+  before_filter :authenticate
+  before_action :set_cafe_work, only: [:edit, :show, :update, :destroy, :remove_worker]
+
+  def show
+  end
+
+  def new
+    @cwork = CafeWork.new
+  end
+
+  def edit
+  end
+
+  def create
+    @cwork = CafeWork.new(c_w_params)
+    flash[:notice] = 'Cafejobbet skapades, success.' if @cwork.save
+    respond_with @cwork
+  end
+
+  def update
+    if @cwork.update(c_w_params)
+      flash[:notice] = 'Cafejobbet uppdaterades'
+      redirect_to @cwork
+    else
+      render action: :edit
+    end
+  end
+
+  def destroy
+    # id?
+    @id = @cwork.id
+    @cwork.destroy
+    respond_to do |format|
+      format.html { redirect_to :hilbert, notice: 'Cafepasset raderades.' }
+      format.js
+    end
+  end
+
+  def remove_worker
+    if !@cwork.clear_worker
+      render action: show, notice: "Lyckades inte"
+    end
+  end
+
+  def setup
+    @cwork = CafeWork.new
+  end
+
+  def setup_preview
+    @cworks = []
+    wday = Time.zone.local(params[:cafe_work]["work_day(1i)"].to_i, params[:cafe_work]["work_day(2i)"].to_i, params[:cafe_work]["work_day(3i)"].to_i, params[:cafe_work]["work_day(4i)"].to_i, 0)
+    lp = params[:cafe_work][:lp]
+    @lv_first = params[:lv_first].to_i
+    @lv_last = params[:lv_last].to_i
+    (@lv_first..@lv_last).each do |week|
+      (0..4).each do
+        @cworks << CafeWork.new(work_day: wday, lp: lp, pass: 1, lv: week, d_year: wday.year)
+        @cworks << CafeWork.new(work_day: wday, lp: lp, pass: 2, lv: week, d_year: wday.year)
+        @cworks << CafeWork.new(work_day: wday+2.hours, lp: lp, pass: 3, lv: week, d_year: wday.year)
+        @cworks << CafeWork.new(work_day: wday+2.hours, lp: lp, pass: 4, lv: week, d_year: wday.year)
+        wday = wday + 1.days
+      end
+      wday = wday + 2.days
+    end
+    @cwork = CafeWork.new(c_w_params)
+    render 'setup'
+  end
+
+  def setup_create
+    wday = Time.zone.local(params[:cafe_work]["work_day(1i)"].to_i, params[:cafe_work]["work_day(2i)"].to_i, params[:cafe_work]["work_day(3i)"].to_i, params[:cafe_work]["work_day(4i)"].to_i, 0)
```

Comment:
Line is too long. [193/100]<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.